qa/sc_evm_test_erc20.py

```diff
@@ -0,0 +1,291 @@
+#!/usr/bin/env python3
+import pprint
+from decimal import Decimal
+
+from SidechainTestFramework.account.ac_use_smart_contract import SmartContract
+from SidechainTestFramework.sc_boostrap_info import SCNodeConfiguration, SCCreationInfo, MCConnectionInfo, \
+    SCNetworkConfiguration, LARGE_WITHDRAWAL_EPOCH_LENGTH
+from SidechainTestFramework.sc_test_framework import SidechainTestFramework
+from test_framework.util import assert_equal, assert_true, start_nodes, websocket_port_by_mc_node_index, \
+    forward_transfer_to_sidechain
+from SidechainTestFramework.scutil import bootstrap_sidechain_nodes, start_sc_nodes, \
+    is_mainchain_block_included_in_sc_block, check_mainchain_block_reference_info, AccountModelBlockVersion, \
+    EVM_APP_BINARY, generate_next_blocks, generate_next_block
+
+"""
+Check an EVM ERC20 Smart Contract.
+
+Configuration: bootstrap 1 SC node and start it with genesis info extracted from a mainchain node.
+    - Mine some blocks to reach hard fork
+    - Create 1 SC node
+    - Extract genesis info
+    - Start SC node with that genesis info
+
+Test:
+    For the smart contract:
+        - Deploy the smart contract without initial data
+        - Check initial minting success (static_call)
+        - Check initial supply (static_call)
+        - Check a successful transfer (static_call + actual tx)
+        - Check a reverting transfer (static_call + actual tx)
+        - Check approval + transferFrom (static_call + actual tx)
+"""
+
+
+def call_addr_uint_fn(node, smart_contract, contract_address, source_addr, addr, uint, static_call, generate_block,
+                      method):
+    if static_call:
+        res = smart_contract.static_call(node, method, addr, uint,
+                                         fromAddress=source_addr,
+                                         gasLimit=10000000, gasPrice=10, toAddress=contract_address)
+    else:
+        res = smart_contract.call_function(node, method, addr, uint,
+                                           fromAddress=source_addr,
+                                           gasLimit=10000000, gasPrice=10, toAddress=contract_address)
+    if generate_block:
+        print("generating next block...")
+        generate_next_blocks(node, "first node", 1)
```

Comment:
just a hint fro future: we have another method to generate a single block: `generate_next_block(node, "first node")`

Reply:
I know, afaik that one is used inside `generate_next_blocks` right? I just wanted to keep block generation mostly uniform throughout the scripts